docker-build.sh

```diff
@@ -7,16 +7,26 @@ DOCKERFILE="$CONTEXT/$2"
 REMOTE_IMAGE_NAME=$3
 EXTRA_CACHE=$4
 
-CACHE_NAME=$(echo $REMOTE_IMAGE_NAME | sed -E 's/(:[^:]+)(@[^@]+)?$//'):cache
+WITHOUT_TAG=$(echo $REMOTE_IMAGE_NAME | sed -E 's/(:[^:]+)(@[^@]+)?$//')
+MAIN_CACHE=${WITHOUT_TAG}:cache
+
+NAMESPACE=${NAMESPACE:-"default"}
```

Comment:
Can we make NAMESPACE a required variable? I'm worried about pushing to default without intending to.